### PR TITLE
New package: py-librosa

### PR DESCRIPTION
**Pull request by glennpj**
This PR adds the py-librosa package, along with new dependency packages
and some updates of existing dependency packages.

- new package: py-audioread
- new package: py-resampy
- new package: py-soundfile
- update package: py-numba
- update package: py-llvmlite

py-numba:
- add updated version
- adjust constraints

py-llvmlite:
- add updated version
- adjust constraints
- fix version specifications for llvm
- add environment function to set PIC

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-numba/package.py [hunk 1]
```diff
@@ -12,18 +12,22 @@ class PyNumba(PythonPackage):
...
+    version('0.48.0', sha256='9d21bc77e67006b5723052840c88cc59248e079a907cc68f1a1a264e1eaba017')
     version('0.40.1', sha256='52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33')
     version('0.35.0', sha256='11564937757605bee590c5758c73cfe9fd6d569726b56d970316a6228971ecc3')
 
+    depends_on('python@3.6:', type=('build', 'run'), when='@0.48.0:')
     depends_on('python@3.3:3.7.9999', type=('build', 'run'), when='@0.40.1:')
     depends_on('python@3.3:3.6.9999', type=('build', 'run'), when='@:0.35.0')
+    depends_on('py-numpy@1.15:', type=('build', 'run'), when='@0.48.0:')
     depends_on('py-numpy@1.10:1.99', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numba/package.py]:
```suggestion
    depends_on('py-setuptools', type=('build', 'run'))
```
`numba/entrypoints.py` has an import of `pkg_resources`, which is provided by `py-setuptools`.

File: var/spack/repos/builtin/packages/py-audioread/package.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class PyAudioread(PythonPackage):
+    """cross-library (GStreamer + Core Audio + MAD + FFmpeg) audio decoding for
+    Python."""
+
+    homepage = "https://github.com/beetbox/audioread"
+    url      = "https://pypi.io/packages/source/a/audioread/audioread-2.1.8.tar.gz"
+
+    version('2.1.8', sha256='073904fabc842881e07bd3e4a5776623535562f70b1655b635d22886168dd168')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-audioread/package.py]: Is `py-pytest-runner` needed at build time? I see:
```python
      setup_requires=[                                                          
          'pytest-runner'                                                       
      ],  
```
in the `setup.py`

File: var/spack/repos/builtin/packages/py-soundfile/package.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    """SoundFile is an audio library based on libsndfile, CFFI and NumPy."""
+
+    homepage = "https://github.com/bastibe/PySoundFile"
+    url      = "https://pypi.io/packages/source/S/SoundFile/SoundFile-0.10.3.post1.tar.gz"
+
+    version('0.10.3.post1', sha256='490cff42650733d1832728b937fe99fa1802896f5ef4d61bcf78cf7ebecb107b')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cffi@1.0:', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-soundfile/package.py]: This dependency is optional, I would either remove it or add a variant (default to False) for it.

File: var/spack/repos/builtin/packages/py-audioread/package.py (hunk 2 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-audioread/package.py]: I think it is for testing. I see no difference in the build with or without it. Should I add it?

File: var/spack/repos/builtin/packages/py-soundfile/package.py (hunk 3 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: According to the docs is depends on numpy. However, I see that it also depends on libsndfile, at least at run time. I have that on my system so missed it but spack needs a package for that.

File: var/spack/repos/builtin/packages/py-audioread/package.py (hunk 2 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-audioread/package.py]: I went ahead and added that dependency just to be complete. The build logs do not show a difference.

File: var/spack/repos/builtin/packages/py-soundfile/package.py (hunk 3 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: Numpy is an extra feature, when you `pip install soundfile` it doesn't use numpy unless you tell it to. And yes, it looks like it has a dependency on the `libsndfile` library.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: Well, it does not link to anything in the traditional sense. The python module will load the libsndfile library via CFFI. That feels more like a run dependency to me.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: I don't know much about CFFI...

Link dependencies add the libraries and headers via the compiler wrappers during build time.
Run dependencies add the dependencies `prefix.bin` to `PATH`

Does CFFI find libraries via PATH or something else?

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: Also, I tested building without the `libsndfile` library, even removing the system library. It builds fine without it being present but it is needed at run time to be able to import `soundfile`.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: The library is found via the `find_library` function of `ctypes.util`.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: On Linux, `find_library` will use `ld`, so as long as `$LD_LIBRARY_PATH` is set by `libsndfile`, py-sound will be able to find the library.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: Would it be better just to use the default dependency type?
```
depends_on('libsndfile')
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-soundfile/package.py]: Nah, the default is build/link, so it's the exact opposite of what you already have

**Pull request merged**